### PR TITLE
Save edited name

### LINKED ISSUE #42: Change gochapter and gomodule command to a single access command

### LINKED ISSUE #43: Change backchapter and backmodule command to a single back command

### LINKED ISSUE #51: Modify module folder name in the file structure after edit command is called

### LINKED ISSUE #52: Modify chapter text file in the file structure after edit command is called

### PR DESCRIPTION
**Pull request by Zhu-Ze-Yu**
Fixes #42 
Fixes #43 
Fixes #51 
Fixes #52

### CODE REVIEW

**Review by woozije — COMMENTED** [review]
Looks good to merge

**Comment by Zhu-Ze-Yu** [comment]
Fixes #51 
Fixes #52

File: src/main/java/commands/EditCommand.java [hunk 1]
```diff
@@ -90,9 +91,13 @@ private void editModule(Ui ui, Access access, Storage storage) throws InvalidInp
...
-            ui.showUnedited(MODULE, module.toString());
-            module.setModuleName(moduleOrChapter);
-            ui.showEdited(MODULE, module.toString());
+            File file = new File(storage.getFilePath() + "/" + module.toString());
+            boolean success = file.renameTo(new File(storage.getFilePath() + "/" + moduleOrChapter));
+            if (success) {
+                ui.showUnedited(MODULE, module.toString());
+                module.setModuleName(moduleOrChapter);
+                ui.showEdited(MODULE, module.toString());
+            }
```
Comment (woozije) [inline src/main/java/commands/EditCommand.java]: Should this be done in Storage?

File: src/main/java/commands/EditCommand.java [hunk 2]
```diff
@@ -103,9 +108,17 @@ private void editChapter(Ui ui, Access access, Storage storage) throws InvalidIn
...
+                    + "/" + access.getModule()
+                    + "/" + chapter.toString() + ".txt");
+            boolean success = file.renameTo(new File(storage.getFilePath()
+                    + "/" + access.getModule()
+                    + "/" + moduleOrChapter + ".txt"));
+            if (success) {
+                ui.showUnedited(CHAPTER, chapter.toString());
+                chapter.setChapterName(moduleOrChapter);
+                ui.showEdited(CHAPTER, chapter.toString());
+            }
```
Comment (woozije) [inline src/main/java/commands/EditCommand.java]: Should this be done in storage?

File: src/main/java/storage/Storage.java [hunk 3]
```diff
@@ -171,4 +172,20 @@ public boolean deleteDirectory(File directoryToBeDeleted) {
         }
         return directoryToBeDeleted.delete();
     }
+
+    public boolean renameChapter(String newChapterName, Access access, Storage storage, Chapter chapter) {
```
Comment (Jane-Ng) [inline src/main/java/storage/Storage.java]: Do not need to pass in Storage object

File: src/main/java/storage/Storage.java [hunk 4]
```diff
@@ -171,4 +172,20 @@ public boolean deleteDirectory(File directoryToBeDeleted) {
...
+        File file = new File(storage.getFilePath()
+                + "/" + access.getModule()
+                + "/" + chapter.toString() + ".txt");
+        boolean success = file.renameTo(new File(storage.getFilePath()
+                + "/" + access.getModule()
+                + "/" + newChapterName + ".txt"));
+        return success;
+    }
+
+    public boolean renameModule(String newModuleName, Storage storage, Module module) {
```
Comment (Jane-Ng) [inline src/main/java/storage/Storage.java]: Do not need to pass in Storage object

File: src/main/java/storage/Storage.java (hunk 3 above)
Comment (Zhu-Ze-Yu) [reply src/main/java/storage/Storage.java]: you mean the Storage storage?

Comment (Jane-Ng) [reply src/main/java/storage/Storage.java]: Yes

**Comment by Zhu-Ze-Yu** [comment]
Fixes #43

**Comment by Zhu-Ze-Yu** [comment]
Fixes #42

**Comment by Zhu-Ze-Yu** [comment]
Save edited chapter name and module name 
Combine GoChapterCommand and GoModuleCommand 
Combine BackChapterCommand and BackModuleCommand

**Pull request merged**